flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/nonha/embedded/EmbeddedJobResultStore.java

```diff
@@ -67,7 +79,7 @@ public boolean hasDirtyJobResultEntryInternal(JobID jobId) {
 
     @Override
     public boolean hasCleanJobResultEntryInternal(JobID jobId) {
-        return cleanJobResults.containsKey(jobId);
+        return cleanJobResults.asMap().containsKey(jobId);
```

Comment:
```suggestion
        return cleanJobResults.getIfPresent(jobId) != null;
```
nit: There is a interface method that appears to be a better fit to the use case.